### PR TITLE
refactor(step-generation): remove dependency on components

### LINKED ISSUE #7747: JS cleanup: factor out `Mount` type to shared-data
Move `Mount` type from `components` to `shared-data`, then we can remove `components` as a dependency of `step-generation`

This is follow-up work after #7702

### PR DESCRIPTION
**Pull request by IanLondon**
# Overview

Closes #7747 -- should be no more `components` imports into `step-generation`

# Changelog

# Review requests

- Flow & TS pass
- No other `components` import into `step-generation`

# Risk assessment

Low, SG only

### CODE REVIEW

File: step-generation/src/types.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
 // @flow
-import type { Mount } from '@opentrons/components'
 import typeof {
   MAGNETIC_MODULE_TYPE,
   TEMPERATURE_MODULE_TYPE,
   THERMOCYCLER_MODULE_TYPE,
 } from '@opentrons/shared-data'
```
Comment (mcous) [inline step-generation/src/types.js:6]: Looks like `step-generation/package.json` is missing `@opentrons/shared-data` as a dependency

**Review by mcous — APPROVED** [review]
LGTM, but might be advantageous to fix up the package.json::dependencies section while we're here

**Review by Kadee80 — APPROVED** [review]
🧭

File: step-generation/src/types.js (hunk 1 above)
Comment (shlokamin) [reply step-generation/src/types.js:6]: good catch!

**Review by shlokamin — APPROVED** [review]
LGTM, same comment as @mcous re:package.json. that's my bad i forgot to add it in the first pass

**Pull request merged**